instaclustr/resource_cluster.go

```diff
@@ -323,45 +364,56 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
 	d.SetId(id)
 	d.Set("cluster_id", id)
 	log.Printf("[INFO] Cluster %s has been created.", id)
-	return nil
+
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		cluster, err := client.ReadCluster(id)
+
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("[Error] Error retrieving cluster info: %s", err))
+		}
+
+		if cluster.ClusterStatus == "RUNNING" {
+			return resource.NonRetryableError(resourceClusterRead(d, meta))
+		}
+
+		return resource.RetryableError(fmt.Errorf("[Error] Cluster is in state %s", cluster.ClusterStatus))
+	})
 }
 
 func resourceClusterUpdate(d *schema.ResourceData, meta interface{}) error {
 	d.Partial(true)
 	// currently only cluster resize is supported
-	if !d.HasChange("node_size") {
-		return fmt.Errorf("[Error] The cluster doesn't support update")
-	}
+	if d.HasChange("node_size") {
```

Comment:
Would like to forward previous discussions on this change - https://github.com/instaclustr/terraform-provider-instaclustr/pull/51#discussion_r484621963

Reply:
My work is based on #51 because I need `private_contact_addresses` property. Of course, we can track this change in #51.

Reply:
Fair point, will chase it up on #51